doc/man3/OPENSSL_LH_COMPFUNC.pod

```diff
@@ -184,6 +184,12 @@ audit/verify and also opens the window of opportunity for stack
 corruption and other hard-to-find bugs.  It also, apparently, violates
 ANSI-C.
 
+The LHASH code is not thread safe. All updating operations must
+be performed under a write lock. All retrieve operations should be
+performed under a read lock or under a write lock if accuracy in the
+gathered usage statistics is desired. The statistical output functions
+from L<OPENSSL_LH_stats(3)> require a read lock when called.
```

Comment:
I sense certain ambiguity here. Thing is that while read lock is sufficient for OPENSSL_LH_stats itself, it effectively depends on write locks at retrieve time. And I fail to see this relation emphasized. Would something along following be more appropriate? "... under a read lock, *unless* accurate usage statistic is desired. In which case a write lock [for retrieve operations] should be used, for OPENSSL_LH_stats read lock suffices." Idea is to "break" it into two "causality" chains...